File: Java/leetcode75/src/test/java/org/phoenix/leetcode/challenges/Problem35_CountGoodNodesInBinaryTreeTest.java
package org.phoenix.leetcode.challenges;

import org.junit.jupiter.api.Test;
import org.phoenix.leetcode.challenges.Problem33_MaximumDepthOfBinaryTree.TreeNode;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Problem35_CountGoodNodesInBinaryTreeTest {

    private final Problem35_CountGoodNodesInBinaryTree test = new Problem35_CountGoodNodesInBinaryTree();

    @Test
    void goodNodes() {
        TreeNode root = new TreeNode(3);
        root.left = new TreeNode(1);
        root.left.left = new TreeNode(3);
        root.right = new TreeNode(4);
        root.right.left = new TreeNode(1);
        root.right.right = new TreeNode(5);

        assertEquals(4, test.goodNodes(root));
    }
}

File: JavaScript/leetcode75/src/test/Problem35_CountGoodNodesInBinaryTree.test.js
const goodNodes = require("../main/Problem35_CountGoodNodesInBinaryTree");

//  Definition for a binary tree node.
function TreeNode(val, left, right) {
    this.val = val === undefined ? 0 : val;
    this.left = left === undefined ? null : left;
    this.right = right === undefined ? null : right;
  }

test("should return the count of good node", () => {
  let root = new TreeNode(3, 1, 4);
  root.left = new TreeNode(1, 3, null);
  root.left.left = new TreeNode(3, null, null);
  root.right = new TreeNode(4, 1, 5);
  root.right.left = new TreeNode(1, null, null);
  root.right.right = new TreeNode(5, null, null);

  expect(goodNodes(root)).toBe(4);
});
